### PR TITLE
(maint) Enhance VM Pooler developer experience

### PR DESCRIPTION
**Pull request by glennsarti**
Previously, a bundle install would not pull in gems from Gemfile.local or
~/.gemfile which are common development workflows in Puppet.  This commit
modifies the Gemfile to pull in these additional gemfiles if they exist.  This
commit also adds common files and folders to gitignore which should not be
committed to this repository.

Previously, if you ran the vpooler via ruby, pressing Ctrl-C would terminate the
Webserver however the PoolManager does not have a handler and would instead
just keep executing.  This commit adds a global Ctrl-C hook which terminates
both the api and manager threads.  This behaviour will only be enabled if the
`VMPOOLER_DEBUG` environment variable exists so that it does not affect VMPooler
when running in production environments.

### CODE REVIEW

**Comment by glennsarti** [comment]
Needs to fix https://github.com/puppetlabs/vmpooler/blob/master/lib/vmpooler.rb#L28 as well..Should be load_file not just load

**Comment by glennsarti** [comment]
Ready for review

**Comment by mckern** [comment]
Some of the commits in this PR are also in #180 -- is there a specific merge order desired here?

**Comment by glennsarti** [comment]
Yes.. This first then 180.  I added in the PR comments 180 that this builds on 177.  Perhaps that wasn't as obvious as it should be.

File: Gemfile [hunk 1]
```diff
@@ -23,3 +23,13 @@ group :test do
...
 end
+
+# Evaluate Gemfile.local if it exists
+if File.exists? "#{__FILE__}.local"
+  eval(File.read("#{__FILE__}.local"), binding)
+end
+
+# Evaluate ~/.gemfile if it exists
+if File.exists?(File.join(Dir.home, '.gemfile'))
+  eval(File.read(File.join(Dir.home, '.gemfile')), binding)
```
Comment (mckern) [inline Gemfile]: Like I said in #180, I'm not generally a fan of the `Gemfile.local` pattern, but it's not something I feel strongly enough about to push back on.

BUT! You can eschew the passing of `binding` if you use `instance_eval` instead:

```
instance_eval(File.read("#{__FILE__}.local")
```

**Comment by glennsarti** [comment]
@mckern Updated with `instance_eval`

File: vmpooler [hunk 2]
```diff
@@ -26,4 +26,11 @@ manager = Thread.new {
   ).execute!
 }
 
+if ENV['VMPOOLER_DEBUG']
+  trap("INT") {
+    puts "Shutting down."
+    [api, manager].each { |t| t.exit }
+  }
```
Comment (mckern) [inline vmpooler]: Nitpick: multiline blocks should be `do`/`end`, not braces. Not gonna block on this but maybe this repo needs some Rubocop love?

Comment (glennsarti) [inline vmpooler]: Fixed

**Pull request merged**